src/main/java/de/tum/in/www1/artemis/service/FileService.java

```diff
@@ -1052,6 +1052,20 @@ public static String removeIllegalCharacters(String string) {
         return string.replaceAll("[^a-zA-Z0-9.\\-]", "_").replaceAll("\\.+", ".");
     }
 
+    public static String removeIllegalPathCharacters(String string) {
+        // First replace all characters that are not (^) a-z, A-Z, 0-9, '-', '.' with '_'
```

Comment:
the comment does not seem to be correct when looking at the replacements below

Reply:
Fixed in: https://github.com/ls1intum/Artemis/pull/6860/commits/4e28cb44c28f90faf566fc6ff380b2bff8e37c61